cmd/runtime_options_test.go

```diff
@@ -24,246 +24,255 @@ import (
+		expCompatMode: defaultCompatMode,
 	},
-	{
-		"valid env vars with special chars",
-		true,
-		map[string]string{"test1": "value 1"},
-		[]string{"--env", "test2=value,2", "-e", `test3= ,  ,,, value, ,, 2!'@#,"`},
-		false,
-		map[string]string{"test1": "value 1", "test2": "value,2", "test3": ` ,  ,,, value, ,, 2!'@#,"`},
+	"valid env vars with spaces": {
+		useSysEnv:     true,
+		systemEnv:     map[string]string{"test1": "value 1"},
+		cliFlags:      []string{"--env", "test2=value 2"},
+		expEnv:        map[string]string{"test1": "value 1", "test2": "value 2"},
+		expCompatMode: defaultCompatMode,
+	},
+	"valid env vars with special chars": {
+		useSysEnv:     true,
+		systemEnv:     map[string]string{"test1": "value 1"},
+		cliFlags:      []string{"--env", "test2=value,2", "-e", `test3= ,  ,,, value, ,, 2!'@#,"`},
+		expEnv:        map[string]string{"test1": "value 1", "test2": "value,2", "test3": ` ,  ,,, value, ,, 2!'@#,"`},
+		expCompatMode: defaultCompatMode,
 	},
 }
 
-func TestEnvVars(t *testing.T) {
-	for _, tc := range envVarTestCases {
-		t.Run(fmt.Sprintf("EnvVar test '%s'", tc.name), func(t *testing.T) {
-			os.Clearenv()
-			for key, val := range tc.systemEnv {
-				require.NoError(t, os.Setenv(key, val))
-			}
-			flags := runtimeOptionFlagSet(tc.useSysEnv)
-			require.NoError(t, flags.Parse(tc.cliOpts))
+func testRuntimeOptionsCase(t *testing.T, tc runtimeOptionsTestCase) {
+	flags := runtimeOptionFlagSet(tc.useSysEnv)
+	require.NoError(t, flags.Parse(tc.cliFlags))
+
+	rtOpts, err := getRuntimeOptions(flags, tc.systemEnv)
+	if tc.expErr {
+		require.Error(t, err)
+		return
+	}
+	require.NoError(t, err)
+	require.EqualValues(t, tc.expEnv, rtOpts.Env)
+	assert.Equal(t, tc.expCompatMode, rtOpts.CompatibilityMode)
 
-			rtOpts, err := getRuntimeOptions(flags)
-			if tc.expErr {
-				require.Error(t, err)
-				return
-			}
-			require.NoError(t, err)
-			require.EqualValues(t, tc.expEnv, rtOpts.Env)
+	compatMode, err := lib.ValidateCompatibilityMode(rtOpts.CompatibilityMode.String)
+	require.NoError(t, err)
 
-			// Clear the env again so real system values don't accidentally pollute the end-to-end test
-			os.Clearenv()
+	jsCodeLines := make([]string, 0, len(tc.expEnv)+2)
+	if compatMode == lib.CompatibilityModeExtended {
+		jsCodeLines = append(jsCodeLines, "export default function() {")
+	} else {
+		jsCodeLines = append(jsCodeLines, "module.exports.default = function() {")
+	}
 
-			jsCode := "export default function() {\n"
-			for key, val := range tc.expEnv {
-				jsCode += fmt.Sprintf(
-					"if (__ENV.%s !== `%s`) { throw new Error('Invalid %s: ' + __ENV.%s); }\n",
-					key, val, key, key,
-				)
-			}
-			jsCode += "}"
+	for key, val := range tc.expEnv {
+		jsCodeLines = append(jsCodeLines, fmt.Sprintf(
+			"if (__ENV.%s !== `%s`) { throw new Error('Invalid %s: ' + __ENV.%s); }",
+			key, val, key, key,
+		))
+	}
+	jsCodeLines = append(jsCodeLines, "}")
+	jsCode := []byte(strings.Join(jsCodeLines, "\n"))
```

Comment:
Maybe use [bytes.Buffer](https://golang.org/pkg/bytes/#Buffer) or [string.Builder](https://golang.org/pkg/strings/#Builder) ? I think [bytes.Buffer](https://golang.org/pkg/bytes/#Buffer) with [fmt.Fprintf](https://yourbasic.org/golang/build-append-concatenate-strings-efficiently/#before-go-1-10) will be the best from performance(not that it matters all that much in this case and readability :).

Reply:
Fixed by https://github.com/loadimpact/k6/pull/1325/commits/c9d75bd1cb74cafd11182125da5ffe2d8e61b37b